atlante/internal/urlutil/urlutil.go

```diff
@@ -12,38 +12,48 @@ import (
 	"time"
 )
 
+// ErrRemoteFile is an error that was caused when trying to access a remote file.
+// the remote location is provided
 type ErrRemoteFile struct {
 	// Location is the name of the file that was attempted
 	Location *url.URL
 	Err      error
 }
 
+// Error implements the error interface
 func (e ErrRemoteFile) Error() string {
 	return fmt.Sprintf("error obtaining remote file (%v): %v", e.Location, e.Err)
 }
 
+// ErrUnsupportedScheme is returned when the scheme is not understood
 type ErrUnsupportedScheme ErrRemoteFile
 
+// Error implements the error interface
 func (e ErrUnsupportedScheme) Error() string {
 	return fmt.Sprintf("unsupported scheme (%v), for location %v", strings.ToLower(e.Location.Scheme), e.Location)
 }
 
+// ErrFile is retuned when an error related to a local file is return
 type ErrFile struct {
 	// Filename is the name of the file that was attempted
 	Filename string
 	Err      error
 }
 
+// Error implements the error interface
 func (e ErrFile) Error() string {
 	return fmt.Sprintf("error opening local file (%v): %v", e.Filename, e.Err)
 }
 
+// ErrFileNotExists is returned when the requested local file is not found
 type ErrFileNotExists ErrFile
 
+// Error implements the error interface
 func (e ErrFileNotExists) Error() string {
 	return fmt.Sprintf("file at location (%v) not found!", e.Filename)
 }
 
+// ReaderCloser is an io.Reader with a close method
 type ReaderCloser interface {
```

Comment:
Why not use the native io.ReadCloser interface? https://golang.org/pkg/io/#ReadCloser

Reply:
I think I just did not see it when I looked. I remember being confused that there wasn't one...